### PR TITLE
[Move] Implement Electrify

### PR DESCRIPTION
**Pull request by innerthunder**
## What are the changes the user will see?
This implements the move [Electrify](https://bulbapedia.bulbagarden.net/wiki/Electrify_(move)).

## Why am I making these changes?
More implementations are good, and this isn't much different from Plasma Fists, which I just implemented.

## What are the changes from a developer perspective?
- `data/battler-tags`: New `ELECTRIFIED` tag converts the tag holder's moves to Electric type for the rest of the turn.
- `data/move`: Electrify now uses an `AddBattlerTagAttr` to add an `ELECTRIFIED` tag to the target.
- `field/pokemon`: `getMoveType` now returns `Type.ELECTRIC` when the Pokemon has an `ELECTRIFIED` tag (this applies after move-type-changing abilities like Pixilate).

## How to test the changes?
`npm run test electrify` (2 new tests)

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by DayKev — APPROVED** [review]

File: src/data/battler-tags.ts [hunk 1]
```diff
@@ -2310,6 +2310,20 @@ export class TarShotTag extends BattlerTag {
   }
 }
 
+/**
+ * Battler Tag implementing
```
Comment (Tempo-anon) [inline src/data/battler-tags.ts:2314]: This doc is incomplete

File: src/data/battler-tags.ts [hunk 2]
```diff
@@ -2310,6 +2310,20 @@ export class TarShotTag extends BattlerTag {
...
+ */
+export class ElectrifiedTag extends BattlerTag {
+  constructor() {
+    super(BattlerTagType.ELECTRIFIED, BattlerTagLapseType.TURN_END, 1, Moves.ELECTRIFY);
+  }
+
+  override onAdd(pokemon: Pokemon): void {
+    // "{pokemonNameWithAffix}'s moves have been electrified!"
+    pokemon.scene.queueMessage(i18next.t("battlerTags:electrifiedOnAdd", { pokemonNameWithAffix: getPokemonNameWithAffix(pokemon) }));
```
Comment (Tempo-anon) [inline src/data/battler-tags.ts:2323]: I assume this is a new locale key being added? Is there a link to the PR adding it in the other repo?

Comment (innerthunder) [reply src/data/battler-tags.ts:2323]: Yeah, it's already been merged in https://github.com/pagefaultgames/pokerogue-locales/pull/17

File: src/data/battler-tags.ts (hunk 1 above)
Comment (innerthunder) [reply src/data/battler-tags.ts:2314]: oops

**Pull request merged**